### PR TITLE
BrowserFragment: Start to replace Session usage with browser-state

### PR DESCRIPTION
**Pull request by pocmo**
Slowly starting to replace `Session` and `SessionManager` usage in Focus.

### CODE REVIEW

File: app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.kt [hunk 1]
```diff
@@ -383,7 +393,7 @@ class BrowserFragment :
     }
 
     private fun initialiseCustomTabUi(view: View) {
-        val customTabConfig = session.customTabConfig!!
+        val customTabConfig = tab.ifCustomTab()!!.config
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.kt:396]:
```suggestion
        val customTabConfig: CustomTabConfig = tab.ifCustomTab()?.config ?: return
```

nit: early return, maybe?

**Review by jonalmeida — APPROVED** [review]
✅

File: app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.kt (hunk 1 above)
Comment (pocmo) [reply app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.kt:396]: Yeah, this whole code is not nice. At this point we already know that it's a custom tab, so having to deal with nullability here is annoying. Hm, actually, maybe I can just pass in the `CustomTabConfig` if the check happens outside, I'll try that and update the PR.

Comment (jonalmeida) [reply app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.kt]: It shows up now, so I fixed the conflict in the PR and we can merge it.

**Pull request merged**